### PR TITLE
Modernise UI

### PR DESCRIPTION
**Pull request by timja**
Requires https://github.com/jenkinsci/jenkins/pull/6307

Principles applied:
- All pages have a heading
- Title matches the heading
- All buttons say what they do instead of OK
- All buttons on a new line not next to text
- All primary buttons are styled appropriately
- Switch to core controls where possible otherwise backport changes

TODO:
- Get upstream PR merged

image

<details>
<summary>Manage Jenkins</summary>

image

</details>

<details>
<summary>Add credential button spacing fixed</summary>

image

</details>

<details>
<summary>Add credential dialog</summary>

image
image

</details>

<details>
<summary>Global credentials list</summary>

image

</details>

<details>
<summary>New Credentials</summary>

image

</details>

<details>
<summary>Update credentials</summary>

image

</details>

<details>
<summary>Move credentials</summary>

image

</details>

<details>
<summary>Delete credentials</summary>

image

</details>

<details>
<summary>Add Domain</summary>

image

</details>

<details>
<summary>New domain</summary>

image

</details>

<details>
<summary>View domain</summary>

image

</details>

<details>
<summary>Update domain</summary>

image

</details>

<details>
<summary>Delete domain</summary>

image

</details>

<details>
<summary>Credentials parameter</summary>

image

</details>

### CODE REVIEW

**Comment by NotMyFault** [comment]
> Not 100% sure about the whole security section having the same icon
> 
> image

Yeah, I'd preserve the lock for the "Configure Global Security" tab and use the key icon here.
Considering we have a key.svg in core, we could serve the icon from there too, thoughts?

**Comment by timja** [comment]
yeah sure

**Comment by timja** [comment]
2 issues I'm aware of atm:

1. Icon on add credential dialog is giant, will prob remove once I find
2. User credential provider Icon is showing the missing ionicon for some reason

**Comment by timja** [comment]
Done all I plan to do, merging is blocked until https://github.com/jenkinsci/jenkins/pull/6307 is released.

But I would appreciate reviews and feedback

File: src/main/resources/lib/credentials/select/select.css [hunk 1]
```diff
@@ -5,19 +5,45 @@ div#credentialsDialog {overflow-y: scroll;}
...
-    margin-bottom: 1em;
+    position: relative;
+    padding: 1rem;
+    margin: 1rem 0;
     word-break: break-word;
+    border-radius: 6px;
+    z-index: 0;
 }
+
+.help-content::before {
```
Comment (NotMyFault) [inline src/main/resources/lib/credentials/select/select.css:17]: What is that used for? Can't we rely on `jenkins-help-button` or `l:helpIcon` from core?

**Review by NotMyFault — COMMENTED** [review]
- The dropdown to add a new credentials domain on the "Store" column solely displays the chevron but not the context menu to perform the operation.

- Now we have three icon sources in place, I'm wondering if it would be worth to replace/overwrite the table icons too 🤔 
|   |   |

File: src/main/resources/lib/credentials/select/select.css (hunk 1 above)
Comment (timja) [reply src/main/resources/lib/credentials/select/select.css:17]: It’s used when you add a credential parameter to a job.

credentials plugins has created a ‘simple help’ that can have inline text. Can maybe refactor it to use core in some way, this is just fixing it up to look right.

**Comment by timja** [comment]
Sure can try that what are the symbols for each ?

**Comment by NotMyFault** [comment]
> Sure can try that what are the symbols for each ?

Ribbon for certificate.svg (overwriting in core), symbol-jenkins for system-scope.svg, symbol-key for credential.svg (overwriting in core) and I added symbol-id-card for the user pass one.

If you don't mind, you can simply apply my patches I used to test it with https://gist.github.com/NotMyFault/daeaeda507eb1c62dacc3bcfbbeb58c0

**Comment by timja** [comment]
Looks good on the view credentials table:
image


Bit weird here because of the blue:

image

**Comment by NotMyFault** [comment]
I guess the blue comes from core? Spots like dropdowns still use it, e.g.

**Comment by timja** [comment]
Looks like because it's a link for some reason =/

File: pom.xml [hunk 2]
```diff
@@ -67,7 +67,7 @@
   <properties>
     <changelist>999999-SNAPSHOT</changelist>
     <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
-    <jenkins.version>2.308</jenkins.version>
+    <jenkins.version>2.338-rc32251.e9c10b_a_26b_b_0</jenkins.version>
```
Comment (timja) [inline pom.xml:70]: needs to be adjust prior to merge

File: src/main/resources/com/cloudbees/plugins/credentials/CredentialsStoreAction/DomainWrapper/newCredentials.jelly [hunk 3]
```diff
@@ -63,13 +64,13 @@
                   </f:dropdownListBlock>
                 </j:forEach>
               </f:dropdownList>
-              <!--f:dropdownDescriptorSelector field="credentials" title="${%Kind}" lazy="it"/-->
+              <!--<f:dropdownDescriptorSelector field="credentials" title="${%Kind}" descriptors="${descriptors}"/> -->
```
Comment (timja) [inline src/main/resources/com/cloudbees/plugins/credentials/CredentialsStoreAction/DomainWrapper/newCredentials.jelly:67]: After making these changes I _think_ it works but didn't want to do it in the same PR

File: src/main/java/com/cloudbees/plugins/credentials/CredentialsStoreAction.java [hunk 4]
```diff
@@ -298,7 +298,7 @@ public ContextMenu getContextMenu(String prefix) {
         ContextMenu menu = new ContextMenu();
         if (getStore().isDomainsModifiable() && getStore().hasPermission(MANAGE_DOMAINS)) {
             menu.add(ContextMenuIconUtils.buildUrl(prefix, "newDomain"),
-                    getMenuItemIconUrlByClassSpec("icon-credentials-new-domain icon-md"),
```
Comment (timja) [inline src/main/java/com/cloudbees/plugins/credentials/CredentialsStoreAction.java:301]: it's this line here that breaks the context menu loading for add domain.

I think it's using the icon to determine the url =/.

Needs refactoring

**Comment by timja** [comment]
I've left out changes to the context menu for two reasons:
1. I wasn't able to figure out how to apply a symbol to it short of setting the `iconXml` field which requires me to load a symbol in the code here which requires calling a restricted API (I disabled access restrictor and verified that works)
1. Translations don't seem to apply to it, I tried using an Icon that has a translation but the context menu code path doesn't seem to go through that (I wasn't able to track down how it's actually built but I didn't spend too long tracing it).

FYI @janfaracik if you have any ideas but probably not critical for the initial PR

File: src/main/java/com/cloudbees/plugins/credentials/CredentialsStoreAction.java [hunk 5]
```diff
@@ -58,6 +58,7 @@
 import hudson.util.HttpResponses;
 import hudson.util.Secret;
 import hudson.util.XStream2;
+import java.awt.Menu;
```
Comment (timja) [inline src/main/java/com/cloudbees/plugins/credentials/CredentialsStoreAction.java:61]:
```suggestion
```

File: pom.xml (hunk 2 above)
Comment (timja) [reply pom.xml:70]:
```suggestion
    <jenkins.version>2.340</jenkins.version>
```

File: pom.xml [hunk 6]
```diff
@@ -67,7 +67,7 @@
   <properties>
     <changelist>999999-SNAPSHOT</changelist>
     <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
-    <jenkins.version>2.319.1</jenkins.version>
+    <jenkins.version>2.340</jenkins.version>
```
Comment (timja) [inline pom.xml:70]: FWIW intentionally not updating the bom artifact ID right now to reduce conflicts with dependabot

**Review by NotMyFault — APPROVED** [review]
Looks amazing 🎉

**Comment by jglick** [comment]
@timja is this & #283 tested and ready to release?

File: pom.xml (hunk 6 above)
Comment (jglick) [reply pom.xml:70]: WDYM? Should it not be set to `weekly`? Or at least `2.332.x`?

**Review by jglick — APPROVED** [review]
Seems OK though I do not follow details. @timja go ahead and merge if you like; seems you have permissions.

**Comment by timja** [comment]
> Seems OK though I do not follow details. @timja go ahead and merge if you like; seems you have permissions.

I'm not a maintainer here, org owner gives me the green tick though.

**Comment by timja** [comment]
Builds green now @jglick

**Comment by daniel-beck** [comment]
May have caused [JENKINS-68791](https://issues.jenkins.io/browse/JENKINS-68791)

File: src/main/java/com/cloudbees/plugins/credentials/CredentialsProvider.java [hunk 7]
```diff
@@ -1075,7 +1075,7 @@ public String getDisplayName() {
      */
     @Override
     public String getIconClassName() {
-        return "icon-credentials-credentials";
+        return "symbol-key";
```
Comment (basil) [inline src/main/java/com/cloudbees/plugins/credentials/CredentialsProvider.java:1078]: I think the `wallet` icon would have been a better choice for the reasons explained in [JENKINS-69775](https://issues.jenkins.io/browse/JENKINS-69775).

**Pull request merged**